wallet/how-to/use-non-evm-networks/starknet/manage-starknet-networks.md

```diff
@@ -0,0 +1,140 @@
+
+  ```javascript
+  const checkCurrentNetwork = async () => {
+    if (typeof window.ethereum !== "undefined" && window.ethereum.isMetaMask) {
+      try {
+        const response = await window.ethereum.request({
+          method: "wallet_invokeSnap",
+          params: {
+            snapId: "npm:@consensys/starknet-snap",
+            request: {
+              method: "starknet_getChainId"
+            }
+          }
+        });
+        
+        let networkName;
+        switch (response) {
+          case "0x534e5f4d41494e":
+            networkName = "Mainnet";
+            break;
+          case "0x534e5f5345504f4c4941":
+            networkName = "Sepolia Testnet";
+            break;
+          default:
+            networkName = "Unknown Network";
+        }
+        
+        console.log("Currently connected to:", networkName);
+        return response;  // Returns the chain ID.
+      } catch (error) {
+        console.error("Error getting current Starknet network:", error);
+        throw error;
+      }
+    } else {
+      console.error("MetaMask not detected or Snaps not supported");
+      throw new Error("MetaMask not detected or Snaps not supported");
+    }
+  };
+
+  checkCurrentNetwork().then(chainId => {
+    console.log("Chain ID:", chainId);
+  }).catch(error => {
+    console.error("Error:", error);
+  });
+  ```
+
+  </TabItem> 
+</Tabs>
+  
+## Switch networks
+
+Starknet currently supports two public networks, Mainnet and Sepolia testnet.
+Prompt users to switch between networks by setting the
+[chain ID](../../../reference/non-evm-apis/starknet-snap-api.md#supported-networks) of the target network:
+
+<Tabs>
+  <TabItem value="get-starknet" default>
+
+  ```javascript
+  const switchChain = async (chainId: string) => {
+    try {
+      await wallet?.request({
+        type: "wallet_switchStarknetChain",
+        params: { chainId: chainId },
+      });
+        console.log(`Switched to chainId: ${chainId}`);
+    } catch (e) {
+      console.error("Failed to switch chain:", e);
+    }
+  };
+  ```
+
+  </TabItem>
+  <TabItem value="wallet_invokeSnap">
+
+  ```javascript
+  const switchStarknetNetwork = async (chainId) => {
+    if (typeof window.ethereum !== "undefined" && window.ethereum.isMetaMask) {
+      try {
+        await window.ethereum.request({
+          method: "wallet_invokeSnap",
+          params: {
+            snapId: "npm:@consensys/starknet-snap",
+            request: {
+              method: "wallet_switchStarknetChain",
+              params: { chainId: chainId }
+            }
+          }
+        });
+        console.log(`Switched to Starknet network with chainId: ${chainId}`);
+      } catch (error) {
+        console.error("Error switching Starknet network:", error);
+        throw error;
+      }
+    } else {
+      console.error("MetaMask not detected or Snaps not supported");
+      throw new Error("MetaMask not detected or Snaps not supported");
+    }
+  };
```

Comment:
```suggestion
  const switchStarknetNetwork = async (chainId) => {
    const provider = await getEip6963Provider()

    if (provider) {
      try {
        await provider.request({
          method: "wallet_invokeSnap",
          params: {
            snapId: "npm:@consensys/starknet-snap",
            request: {
              method: "starkNet_switchNetwork",
              params: { chainId: chainId }
            }
          }
        });
        console.log(`Switched to Starknet network with chainId: ${chainId}`);
      } catch (error) {
        console.error("Error switching Starknet network:", error);
        throw error;
      }
    } else {
      console.error("MetaMask not detected or Snaps not supported");
      throw new Error("MetaMask not detected or Snaps not supported");
    }
  };
```

Reply:
@stanleyyconsensys are we using `window.ethereum` or EIP-6963 ? I don't think we should mix the methods... unless it is absolutely necessary. I think you mentioned that implementing EIP-6963 is a future task and is not fully implemented. Maybe we should wait, instead of partially introducing EIP-6963 for this. WDYT?